accepted/future-releases/0546-patterns/feature-specification.md

```diff
@@ -1227,6 +1236,59 @@ of an expression statement. This is similar to existing restrictions on map
 literals appearing in expression statements. In the rare case where a user
 really wants one there, they can parenthesize it.
 
+#### Function expression in guard ambiguity
+
+Function expressions also use `=>`, which leads to a potential ambiguity:
+
+```dart
+switch (obj) {
+  _ when a + (b) => (c) => body
+};
+```
+
+This could be interpreted as either:
+
+```dart
+switch (obj) {
+  _ when (a + (b)) => ((c) => body)
+  //     ---------    -------------
+};
+
+switch (obj) {
+  _ when (a + (b) => (c)) => (body)
+  //     ----------------    ------
+};
+```
+
+A similar ambiguity exists with function expressions in initializer lists, if
+the constructor happens to be a factory constructor with `=>` for its body. We
+resolve the ambiguity similarly here:  When `=>` is encountered after `when`, if
```

Comment:
Can you add a sentence or two clarifying whether this applies to just `when` clauses inside switch expressions, or to `when` clauses in general?  E.g. is this allowed?

```dart
f(x, y) {
  switch (x) {
    case _ when y + () => z:
      break;
  }
}
```
(and a similar question for if-case).

Note: See discussion at https://github.com/dart-lang/language/issues/2672#issuecomment-1333968549

Reply:
Done.